### PR TITLE
Generate validation set on data set load

### PR DESCRIPTION
**Pull request by mrkrause**
Just adds an option for generating the validation set during load (mostly handy for early stopping experiment, but...)

Something screwy happens if you set stratify=True in train_test_split, but i can't figure out if it's my fault or not. 

EDIT: I am a moron.

### CODE REVIEW

File: src/mnist_loader.py [hunk 1]
```diff
@@ -112,9 +143,12 @@ def _load_label(filename):
     return data
 
 
+
+
 def main():
     """Totally useless main."""
     (train_iamges, train_labels, t10k_images, t10k_labels) = load_mnist()
+    (train_iamges, train_labels, val_images, val_labels, test_imgages, test_labels) = load_mnist(validation_proportion=0.1)
```
Comment (rfratila) [inline src/mnist_loader.py]: I think you misspelled images here

File: src/mnist_loader.py [hunk 2]
```diff
@@ -35,24 +44,40 @@ def load_fashion_mnist():
...
+    Args:
+        validation_proportion: If not None, extract this much data from the training set and return it as a separate
+            validation set
+        validation_seed: seed/state for selecting validation data
+
+
+    :return: (train_images, train_labels, test_images, test_labels) if validation_proportion=None or zero, otherwise
+        returns (trainimages, train_labels, val_images, val_labels, test_images, test_labels)
     """
+
```
Comment (rfratila) [inline src/mnist_loader.py]: You should make a check that validation_proportion is between [0,1]

File: src/mnist_loader.py (hunk 1 above)
Comment (rfratila) [inline src/mnist_loader.py]: Is this going to break existing calls to mnist_loader? As in, defaulting it to return a validation set will return two extra arguments that certain training scripts are not currently ready to handle. Would you mind updating the scripts or have a way to pass to this the validation_proportion so that it doesn't return the validation set by accident?

**Comment by rfratila** [comment]
I'm not sure if the problem with the stratify parameter is due to the scikit-learn version but can you double check that it's not this? https://stackoverflow.com/questions/34842405/parameter-stratify-from-method-train-test-split-scikit-learn

File: src/mnist_loader.py (hunk 2 above)
Comment (mrkrause) [reply src/mnist_loader.py]: train_test_split throws a ValueError if it's out of range already but I can add it if you want.

File: src/mnist_loader.py (hunk 1 above)
Comment (mrkrause) [reply src/mnist_loader.py]: Nope! I can't decide which is better
1.  Returning validation stuff as (like train_img, train_label, test_img test_label, val_img, val_label), which is a little weird but doesn't change the first four args if no val set is requested , or
2. Returning them in the normal order (train_img, train_label, val_img, val_label, test_img test_label) or  (train_img, train_label, test_img test_label), depending. 

Neither should break anything. I had #2, but switched it to #1.

**Comment by mrkrause** [comment]
Also, free to kill this entire branch if you don't want it. It's mildly helpful for messing around, but it's not amazing or anything :-)

File: src/mnist_loader.py (hunk 2 above)
Comment (rfratila) [reply src/mnist_loader.py]: that's alright then. no need to do the same check.

**Comment by rfratila** [comment]
If you think it is important in helping you implement early stopping functionality then it's probably necessary.
